builtin/mainmenu/tab_simple_main.lua

```diff
@@ -87,97 +100,102 @@ end
-					core.setting_set("address",address)
-					core.setting_set("remote_port",port)
+				gamedata.fav = false
+				local favs = core.get_favorites("local")
+				local fav = menudata.favorites[event.row]
+				local address = fav.address
+				local port    = fav.port
+				gamedata.serverdescription = fav.description
+
+				for i = 1, #favs do
+					if fav.address == favs[i].address and
+							fav.port == favs[i].port then
+						gamedata.fav = true
+					end
 				end
 
+				if address and port then
+					core.setting_set("address", address)
+					core.setting_set("remote_port", port)
+				end
 				tabdata.fav_selected = event.row
 			end
+			return true
 		end
-		return true
 	end
 
-	if fields["cb_public_serverlist"] ~= nil then
-		core.setting_set("public_serverlist", fields["cb_public_serverlist"])
+	if fields.btn_delete_favorite then
+		local current_favourite = core.get_table_index("favourites")
+		if not current_favourite then return end
 
-		if core.setting_getbool("public_serverlist") then
-			asyncOnlineFavourites()
-		else
-			menudata.favorites = core.get_favorites("local")
-			menudata.favorites_is_public = false
-		end
+		core.delete_favorite(current_favourite)
+		asyncOnlineFavourites()
+		tabdata.fav_selected = nil
+
+		core.setting_set("address", "")
+		core.setting_set("remote_port", "30000")
 		return true
 	end
 
-	if fields["cb_creative"] then
-		core.setting_set("creative_mode", fields["cb_creative"])
+	if fields.cb_creative then
+		core.setting_set("creative_mode", fields.cb_creative)
 		return true
 	end
 
-	if fields["cb_damage"] then
-		core.setting_set("enable_damage", fields["cb_damage"])
+	if fields.cb_damage then
+		core.setting_set("enable_damage", fields.cb_damage)
 		return true
 	end
 
-	if fields["btn_mp_connect"] ~= nil or
-		fields["key_enter"] ~= nil then
-
-		gamedata.playername		= fields["te_name"]
-		gamedata.password		= fields["te_pwd"]
-		gamedata.address		= fields["te_address"]
-		gamedata.port			= fields["te_port"]
+	if fields.btn_mp_connect or fields.key_enter then
+		gamedata.playername = fields.te_name
+		gamedata.password   = fields.te_pwd
+		gamedata.address    = fields.te_address
+		gamedata.port	    = fields.te_port
 
 		local fav_idx = core.get_textlist_index("favourites")
 
-		if fav_idx ~= nil and fav_idx <= #menudata.favorites and
-			menudata.favorites[fav_idx].address == fields["te_address"] and
-			menudata.favorites[fav_idx].port    == fields["te_port"] then
+		if fav_idx and fav_idx <= #menudata.favorites and
+			menudata.favorites[fav_idx].address == fields.te_address and
+			menudata.favorites[fav_idx].port    == fields.te_port then
 			local fav = menudata.favorites[fav_idx]
 
 			gamedata.servername        = fav.name
 			gamedata.serverdescription = fav.description
 
 			if menudata.favorites_is_public and
-						not is_server_protocol_compat_or_error(
-							fav.proto_min, fav.proto_max) then
+					not is_server_protocol_compat_or_error(
+						fav.proto_min, fav.proto_max) then
```

Comment:
No, since it belongs to the parenthesis and needs to be distinct from the binary operator breakage.